gnd/src/main/java/com/google/android/gnd/system/MediaUploadManager.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.system;
```

Comment:
Since this class interacts with the remote data store, and not Android SDK specifically, perhaps it belongs in ..persistence.remote? Also, I'd name it something Firestore specific. Later we'll need to extract a generic interface so that we have an internal API to switch providers if necessary.

Reply:
Done